### ISSUE TITLE
Add option for display - window - stretch - aspect - expand to expand from not just top left corner

### ISSUE THREAD

**Issue by ghost** [godotengine/godot-proposals#2609]
### Describe the project you are working on
Not working on a project, just noticing this question multiple times

### Describe the problem or limitation you are having in your project
Expand mode only expands from the top left corner. What if I want it to be centered around the viewport?

### Describe the feature / enhancement and how it helps to overcome the problem or limitation
Letting the user set what they want it to expand from will solve this.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams
![image](https://user-images.githubusercontent.com/69117768/114889164-b753c680-9dbe-11eb-9b71-9704eda2af99.png)

### If this enhancement will not be used often, can it be worked around with a few lines of script?
I guess so. Depends on how much "a few" is. You could probably use a viewport with a script to achieve this?

### Is there a reason why this should be core and not an add-on in the asset library?
Resolutions are core.

**Comment by Calinou** [comment]
> Expand mode only expands from the top left corner. What if I want it to be centered around the viewport?

Set your UI nodes' anchor mode to **Center** instead of the default **Top Left**.

**Comment by ShatReal** [comment]
What about non-gui expansion?

**Comment by Calinou** [comment]
> What about non-gui expansion?

2D nodes are already scaled when using the `2d` stretch mode. If you want things to expand from the center, add a Camera2D node, move its origin to be located exactly halfway the project window size (by default, this would be `(512, 300)`) and make sure its **Centered** property is enabled (which is the default).

**Comment by noidexe** [comment]
> What about non-gui expansion?
The simplest solution is to add a Camera2D. That way what gets rendered is always what the camera is focusing on, but I get that it's probably not very intuitive.

**Comment by insomniacUNDERSCORElemon** [comment]
I agree this would be a more useful and intuitive setup. Allowing for things like bumpers or extra details in cutscenes to be more straightforward because the window can be wider or taller but never both.

If there is any use-case for the current behavior (not just that it doesn't matter for UI and cameras), perhaps this should be added as `expanded keep`. (`keep one dimension`?)

**Comment by Calinou** [comment]
> If there is any use-case for the current behavior (not just that it doesn't matter for UI and cameras), perhaps this should be added as expanded keep. (keep one dimension?)

This sounds like the `keep_width` and `keep_height` stretch aspect modes.

**Comment by insomniacUNDERSCORElemon** [comment]
> This sounds like the `keep_width` and `keep_height` stretch aspect modes.

Exactly yes, if you could have *both*. Or as I've said, it acts exactly like `keep` except the black bars are instead rendered. Thus the suggested names.
